packages/components/src/DataList/DataList.tsx

```diff
@@ -132,33 +151,14 @@ function InternalDataList() {
           <InternalDataListSearch />
         </div>
 
-        {headerData && (
-          <DataListHeader
-            layouts={layoutComponents}
-            headerData={headerData}
-            headerVisibility={headerVisibility}
-            mediaMatches={mediaMatches}
-          />
-        )}
+        <DataListHeader />
       </DataListStickyHeader>
 
-      {initialLoading && (
-        <DataListLoadingState
-          headers={headers}
-          layouts={layoutComponents}
-          mediaMatches={mediaMatches}
-        />
-      )}
+      {initialLoading && <DataListLoadingState />}
 
       {showEmptyState && <InternalDataListEmptyState />}
```

Comment:
An improvement that could be made here is to check if we should render these components inside the component itself, as we're doing for DataListHeader, and others.

But, I can also see the point of being more readable this way.

Reply:
Hmmm that's definitely something to think about!

Reply:
I'll merge this first tho